### PR TITLE
decode only the needed data when calculating the required users

### PR DESCRIPTION
**Pull request by ostcar**
Please test this on a test instance where the delegates can not see user names but can see the agenda and change speakers on the agenda

### CODE REVIEW

File: openslides/utils/access_permissions.py [hunk 1]
```diff
@@ -85,16 +85,15 @@ def add_collection_string(
         """
         user_ids: Set[int] = set()
 
-        all_full_data = await element_cache.get_all_full_data()
         for collection_string in collection_strings:
+            full_data = await element_cache.get_collection_full_data(collection_string)
```
Comment (normanjaeckel) [inline openslides/utils/access_permissions.py]: Maybe plese rename this to something like "collection_full_data".

Comment (ostcar) [reply openslides/utils/access_permissions.py]: done

**Comment by ostcar** [comment]
wait, I add some changes...

**Comment by ostcar** [comment]
please test again

**Comment by ostcar** [comment]
ok. One last time...

File: openslides/utils/cache_providers.py [hunk 2]
```diff
@@ -204,6 +209,23 @@ def get_change_id_cache_key(self) -> str:
...
+    async def get_collection_data(
+        self, collection: str, user_id: Optional[int] = None
+    ) -> Dict[bytes, bytes]:
+        """
+        Returns all elements for a collection from the cache.
+        """
+        if user_id is None:
+            cache_key = self.get_full_data_cache_key()
+        else:
+            cache_key = self.get_restricted_data_cache_key(user_id)
```
Comment (normanjaeckel) [inline openslides/utils/cache_providers.py]: Why do you add this second branch/case für restricted data? This pull request does not use it.

Comment (ostcar) [reply openslides/utils/cache_providers.py]: Thats true. I added if for the future. And this is a bad thing to do.

Comment (normanjaeckel) [reply openslides/utils/cache_providers.py]: Do you want to add the fix for the restricted data cache here directly?

Comment (ostcar) [reply openslides/utils/cache_providers.py]: No. If you don't like the different cache keys at this place, I can remove it. But I would prefer to merge it as it is.

**Comment by normanjaeckel** [comment]
@emanuelschuetze This Pull request should be tested (finally) and merged so I can do some steps upon it.

**Pull request merged**